harmonica/tests/test_transformations.py

```diff
@@ -591,6 +592,75 @@ def test_invalid_grid_with_nans(self, sample_potential):
             total_gradient_amplitude(sample_potential)
 
 
+class TestTilt:
+    """
+    Test tilt function
+    """
+
+    def test_against_synthetic(
+        self, sample_potential, sample_g_n, sample_g_e, sample_g_z
+    ):
+        """
+        Test tilt function against the synthetic model
+        """
+        # Pad the potential field grid to improve accuracy
+        pad_width = {
+            "easting": sample_potential.easting.size // 3,
+            "northing": sample_potential.northing.size // 3,
+        }
+        # need to drop upward coordinate (bug in xrft)
+        potential_padded = xrft.pad(
+            sample_potential.drop_vars("upward"),
+            pad_width=pad_width,
+        )
+        # Calculate the tilt and unpad it
+        tilt_grid = tilt(potential_padded)
+        tilt_grid = xrft.unpad(tilt_grid, pad_width)
+        # Compare against g_tilt (trim the borders to ignore boundary effects)
+        trim = 6
+        tilt_grid = tilt_grid[trim:-trim, trim:-trim]
+        g_e = sample_g_e[trim:-trim, trim:-trim] * 1e-5  # convert to SI units
+        g_n = sample_g_n[trim:-trim, trim:-trim] * 1e-5  # convert to SI units
+        g_z = sample_g_z[trim:-trim, trim:-trim] * 1e-5  # convert to SI units
+        g_horiz_deriv = np.sqrt(g_e**2 + g_n**2)
+        g_tilt = np.arctan2(-g_z, g_horiz_deriv)
```

Comment:
Good catch on the `-g_z` 😉

Reply:
I had to plot it to see what was going on. However, I still don't know if all of this is correct, or if I should put the `-` in the `tilt` function and leave `g_z`  untouched in the test 😅 as you didn't complain I assume that this is the correct way

Reply:
Well, if the function needs g_downward then you'd have to change it in the function. If it doesn't matter, then this is fine. But I don't remember the definition of the tilt to know if it needs the downward component or not.

Reply:
I don't know as well. I based my implementation on [the fatiando's legacy](https://legacy.fatiando.org/api/gravmag.transform.html#fatiando.gravmag.transform.tilt).